File: examples/Delete/Callback.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
Uses the default {@link DeleteCallbackI} instance.
"""

import omero
import omero.callbacks

c = omero.client()
s = c.createSession()

try:
    deleteServicePrx = s.getDeleteService()
    dc = omero.api.delete.DeleteCommand("/Image", 1, None)
    deleteHandlePrx = deleteServicePrx .queueDelete([dc])
    cb = omero.callbacks.DeleteCallbackI(c, deleteHandlePrx)

    try:

        try:
            cb.loop(10, 500)
        except omero.LockTimeout:
            print "Not finished in 5 seconds. Cancelling..."
            if not deleteHandlePrx.cancel():
                print "ERROR: Failed to cancel"

        reports = deleteHandlePrx.report()
        r = reports[0]  # We only sent one command
        print "Report:error=%s,warning=%s,deleted=%s" % \
            (r.error, r.warning, r.actualDeletes)

    finally:
        cb.close()

finally:
    c.closeSession()

File: examples/Delete/Callback.java
import omero.LockTimeout;
import omero.ServerError;
import omero.api.ServiceFactoryPrx;
import Glacier2.CannotCreateSessionException;
import Glacier2.PermissionDeniedException;

/**
 * Uses the default {@link DeleteCallbackI} instance.
 */
public class Callback {

    public static void main(String[] args) throws CannotCreateSessionException,
            PermissionDeniedException, ServerError {

        omero.client c = new omero.client();
        ServiceFactoryPrx s = c.createSession();

        try {
            IDeletePrx deleteServicePrx = s.getDeleteService();
            DeleteCommand dc = new DeleteCommand("/Image", 1, null);
            DeleteHandlePrx deleteHandlePrx = deleteServicePrx
                    .queueDelete(new DeleteCommand[] { dc });
            DeleteCallbackI cb = new DeleteCallbackI(c, deleteHandlePrx);
            try {

                cb.loop(10, 500);

                DeleteReport[] reports = deleteHandlePrx.report();
                DeleteReport r = reports[0]; // We only sent one command
                System.out.println(String.format(
                        "Report:error=%s,warning=%s,deleted=%s", r.error,
                        r.warning, r.actualDeletes));

            } catch (LockTimeout lt) {
                System.out.println("Not finished in 5 seconds. Cancelling...");
                if (!deleteHandlePrx.cancel()) {
                    System.out.println("ERROR: Failed to cancel");
                }
            } finally {
                cb.close();
            }

        } finally {
            c.closeSession();
        }

    }

}
